lib/fizzy/cxx20/bit.hpp

```diff
@@ -24,3 +24,41 @@ template <class To, class From>
     return dst;
 }
 }  // namespace fizzy
+
+#ifdef __cpp_lib_bitops
+
+#include <bit>
+
+namespace fizzy
+{
+constexpr int popcount(uint32_t x) noexcept
+{
+    return std::popcount(x);
+}
+
+constexpr int popcount(uint64_t x) noexcept
+{
+    return std::popcount(x);
+}
+}  // namespace fizzy
+
+#else
+
+#include <cstdint>
+
+namespace fizzy
+{
+constexpr int popcount(uint32_t x) noexcept
+{
+    static_assert(sizeof(x) == sizeof(unsigned int));
+    return __builtin_popcount(x);
+}
+
+constexpr int popcount(uint64_t x) noexcept
```

Comment:
These should be inline, not consexpr as `__builtin_popcount` isn't constexpr.

Reply:
Most builtins are constexpr. https://godbolt.org/z/Pnz4ob.

Reply:
Cool, the gcc documentation had no mention of either.

Reply:
Clang actually marks them constexpr: https://clang.llvm.org/docs/LanguageExtensions.html#intrinsics-support-within-constant-expressions

But GCC does not clarify it: https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html